src/sprites/setup.py

```diff
@@ -0,0 +1,241 @@
+import os
+from dataclasses import dataclass
+
+import pygame
+
+from src.enums import Direction, EntityState
+from src.settings import SCALE_FACTOR, CHAR_TILE_SIZE
+from src.support import resource_path
+
+
+@dataclass
+class _AniFrames:
+    frames: list[pygame.Surface]
+    hitbox: pygame.Rect
+
+    def get_frame(self, index: int) -> pygame.Surface:
+        return self.frames[int(index % len(self.frames))]
+
+    def get_hitbox(self) -> pygame.Rect:
+        return self.hitbox
+
+    def __len__(self) -> int:
+        return len(self.frames)
+
+
+EntityAsset = dict[EntityState, dict[Direction, _AniFrames]]
```

Comment:
As we're using Python 3.12, the `type` keyword allows for type alias specification (I did this somewhere in `src.settings`, I believe).

```suggestion
type EntityAsset = dict[EntityState, dict[Direction, _AniFrames]]
```